docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+1 or more filters. The available filters are the following:
+- [`matchArgs`](#arguments-filter)
+- [`matchReturnArgs`](#return-args-filter)
+- [`matchPIDs`](#pids-filter)
+- [`matchBinaries`](#binaries-filter)
+- [`matchNamespaces`](#namespaces-filter)
+- [`matchCapabilities`](#capabilities-filter)
+- [`matchNamespaceChanges`](#namespace-changes-filter)
+- [`matchCapabilityChanges`](#capability-changes-filter)
+- [`matchActions`](#actions-filter)
+
+#### Arguments filter
+
+Arguments filters can be specified under the `matchArgs` field and provide
+filtering based on the value of the function's argument.
+
+In the next example, a selector is defined with a `matchArgs` filter that tells
+the BPF code to process only the function call for which the second argument,
+index equal to 1, concerns the file under the path `/etc/passwd` or
+`/etc/shadow`. It's using the operator `Equal` to match against the value of
+the argument.
+
+Note that conveniently, we can match against a path directly when the argument
+is of type `file`.
+
+```yaml
+selectors:
+- matchArgs:
+  - index: 1
+  operator: "Equal"
+  values:
+  - "/etc/passwd"
+  - "/etc/shadow"
+```
+
+**Further examples**
+
+In the previous example, we used the operator `Equal`, but we can also use the
+`Prefix` operator and match against all files under `/etc` with:
+```yaml
+selectors:
+- matchArgs:
+  - index: 1
+  operator: "Prefix"
+  values:
+  - "/etc"
+```
+In this situation, an event will be created every time a process tries to
+access a file under `/etc`.
+
+Although it makes less sense, you can also match over the first argument, to
+only detect events that will use the file descriptor 4, which is usually the
+first that come afters stdin, stdout and stderr in process. And combine that
+with the previous example.
+
+```yaml
+- matchArgs:
+  - index: 0
+    operator: "Equal"
+    values:
+    - "3"
+  - index: 1
+    operator: "Prefix"
+    values:
+    - "/etc"
+```
+
+#### Return args filter
+
+Arguments filters can be specified under the `returnMatchArgs` field and
+provide filtering based on the value of the function return value. It allows
+you to filter on the return value, thus success, error or value returned by a
+kernel call.
+
+```yaml
+matchReturnArgs:
+- operator: "NotEqual"
+  values:
+  - 0
+```
+
+#### PIDs filter
+
+PIDs filters can be specified under the `matchPIDs` field and provide filtering
+based on the value of host pid of the process. For example, the following
+`matchPIDs` filter tells the BPF code that observe only hooks for which the
+host PID is equal to either `pid1` or `pid2` or `pid3`:
+
+```yaml
+- matchPIDs:
+  - operator: "In"
+    followforks: true
+    values:
+    - "pid1"
+    - "pid2"
+    - "pid3"
+```
+
+**Further examples**
+
+Another example can be to collect all processes not associated with a
+container's init PID, which is equal to 1. In this way, we are able to detect if
```

Comment:
I think we need another filter to exclude host processes, no? So don't observe host processes + container init PID. May be:
```
- matchPIDs:
  - operator: NotIn
    followforks: false
    isNamespacePID: true
    values:
    - 1
    - 0
```
But would be great to double check.

Reply:
I'm not sure to understand why to add PID 0, I think PID 0 which is the scheduler iirc is not really exposed to userspace and is a kernel thing?